packages/contracts-bedrock/contracts/deployment/FreshSystemDictator.sol

```diff
@@ -0,0 +1,107 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.15;
+
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+import { ProxyAdmin } from "../universal/ProxyAdmin.sol";
+import { L2OutputOracle } from "../L1/L2OutputOracle.sol";
+import { OptimismPortal } from "../L1/OptimismPortal.sol";
+import { L1CrossDomainMessenger } from "../L1/L1CrossDomainMessenger.sol";
+import { L1StandardBridge } from "../L1/L1StandardBridge.sol";
+import { OptimismMintableERC20Factory } from "../universal/OptimismMintableERC20Factory.sol";
+
+/**
```

Comment:
The flow of this contract makes sense to me. Do all of the proxy upgrades in a single step, then we have time to do an offchain check that they worked as expected. We can call `step1` again if anything went wrong. 

One failure case that sticks out to me is if we deploy this contract with the incorrect config. There is no way out of a misconfigured `finalOwner` without redeploying the entire set of contracts to L1. Maybe this isn't a big deal with the fresh dictator since we will not be running it in prod and op-chains should get their config correct at deploy time